crates/red_knot_test/src/lib.rs

```diff
@@ -230,13 +251,15 @@ fn run_test(
             src_roots: vec![src_path],
             extra_paths: configuration.extra_paths().unwrap_or_default().to_vec(),
             custom_typeshed: custom_typeshed_path.map(SystemPath::to_path_buf),
-            python_path: PythonPath::KnownSitePackages(
-                configuration
-                    .python()
-                    .into_iter()
-                    .map(SystemPath::to_path_buf)
-                    .collect(),
-            ),
+            python_path: configuration
+                .python()
+                .map(|sys_prefix| {
+                    PythonPath::SysPrefix(
+                        sys_prefix.to_path_buf(),
+                        SysPrefixPathOrigin::PythonCliFlag,
```

Comment:
I only noticed this now. The `PythonCliFlag` variant isn't very accurate because the source is either `environment.python` or `--python`. But that's unrelated to this PR